tests/v2/validation/schemas/schemas_test.go

```diff
@@ -0,0 +1,322 @@
+}
+
+func (sc *SchemaChangesTestSuite) TestCRDCreateDeleteOperation() {
+	subSession := sc.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a new Custom Resource Definition (CRD).")
+	err := createCRD(sc.client, crdCreateFilePath)
+	require.NoError(sc.T(), err)
+
+	log.Info("Access the schemas endpoint for the newly created CRD and verify that it can be accessed successfully without any errors.")
+	err = kwait.PollUntilContextTimeout(context.Background(), twoSecondTimeout, defaults.FiveSecondTimeout, false, func(ctx context.Context) (done bool, pollErr error) {
+		_, pollErr = getSchemaByID(sc.client, localCluster, customSchemaID)
+		if pollErr != nil {
+			return false, pollErr
+		}
+		return true, nil
+	})
+	schemaResponse, err := getSchemaByID(sc.client, localCluster, customSchemaID)
+	require.NoError(sc.T(), err)
+
+	log.Info("Verify that the schema's resourceFields field is set to null.")
+	id := schemaResponse["id"].(string)
+	require.Nil(sc.T(), schemaResponse["resourceFields"], "ResourceFields should be null for schema "+id)
+
+	log.Info("Verify that the schemas endpoint for the new CRD does not list the child schemas and trying to access any of the child schemas results in a '404 Not Found' error.")
+	childSchemas := getChildSchemasForCronTab()
+	err = checkChildSchemasNotListed(sc.client, localCluster, childSchemas, schemaResponse)
+	require.NoError(sc.T(), err)
+
+	log.Info("Verify that the data returned from the schemas endpoint has the resourceMethods and collectionMethods fields.")
+	require.NotNil(sc.T(), schemaResponse["resourceMethods"], "resourceMethods should be present in this schema")
+	require.NotNil(sc.T(), schemaResponse["collectionMethods"], "collectionMethods should be present in this schema")
+
+	log.Info("Access the schema definition for the new CRD and verify that it can be accessed successfully without any errors.")
+	err = kwait.PollUntilContextTimeout(context.Background(), twoSecondTimeout, defaults.FiveSecondTimeout, false, func(ctx context.Context) (done bool, pollErr error) {
+		_, pollErr = getSchemaDefinitionByID(sc.client, localCluster, customSchemaID)
+		if pollErr != nil {
+			return false, pollErr
+		}
+		return true, nil
+	})
+	schemaResponse, err = getSchemaDefinitionByID(sc.client, localCluster, customSchemaID)
+	require.NoError(sc.T(), err)
+
+	log.Info("Verify that the data returned has definitions with resourceFields for each definition type, and that the resourceMethods and collectionMethods fields are present. Also, ensure that fields of type 'integer' in the YAML are parsed as 'int' in the schema definitions.")
+	expectedDefinitions := getChildSchemasForCronTab()
+	err = checkExpectedDefinitions(expectedDefinitions, schemaResponse)
+	require.NoError(sc.T(), err)
+
+	log.Info("Delete the CRD.")
+	err = customresourcedefinitions.DeleteCustomResourceDefinition(sc.client, localCluster, "", customCrdName)
```

Comment:
Why don't we separate out create and delete? I initially thought delete might be using some info from the created CRDs but looking at the test now, it seems we are defining all the vars as constants, if we separate out delete and create, its easier to understand. Any reason why we want to have both tests as a singleton test?

Reply:
Since the CRD must be created before it can be deleted, I initially combined both operations in the same test to avoid duplication. However, I believe it makes more sense to separate them into distinct tests.